jax/experimental/sparse/transform.py

```diff
@@ -353,11 +353,12 @@ def eval_sparse(
 ) -> Sequence[SparsifyValue]:
   env : Dict[core.Var, SparsifyValue] = {}
 
-  def read(var: core.Var) -> Union[Array, SparsifyValue]:
+  def read(var: Union[core.Var, core.Literal]) -> SparsifyValue:
```

Comment:
I think `core.Atom` is an alias for `Union[core.Var, core.Literal]`. Could use it instead.

Reply:
Done